### DISCUSSION TITLE
API: Endpoint for next Status

### DISCUSSION THREAD

**Discussion by bendix-dev** [Traewelling/traewelling#2049] [Ideas / Feature Requests]
I would like to have an endpoint similar to the active status endpoint where you already get a result before the exact second the journey starts as many users already check in a little before the departure (especially at starting stations). 

This could be an extension to the existing active status endpoint (i.e. adding a parameter to control the "margin" you would like to already get a status for). Alternatively I imagine an additional endpoint that just returns the active or next status, completely ignoring how far in the future that journey is happening.

I would also be fine with a different implementation as long as I can get the active, or if there isn't one the next status the user has.

**Comment by HerrLevin** [comment]
Have you tried the `/dashboard/future` endpoint? https://traewelling.de/api/documentation#/Dashboard/getFutureDashboard
Maybe this already suits your needs. 

If so, feel free to close this issue. If not, we'll look further into it. :)

(Context: We're hiding all next checkins that are more than 20 minutes into the future, so that the dashboards won't get too cluttered. This endpoint only shows the future checkins of the currently authenticated user, so that they can see their own checkins but nobody else can. It doesn't entirely fit your description but maybe it'll work for your use case?)

**Comment by bendix-dev** [comment]
I would prefer to only have to make one request which responds with either the active or, if that doesn't exist, a future status.
